### PR TITLE
Provide information for incoming call screens

### LINKED ISSUE #4931: Incoming call screen
Make sure the apis provide enough information for a incoming call screen.

## Solution

Description | Screenshot
---|---
When a call is placed, the application will always display information regarding the call status and whether it is an audio or video call, when technically possible. For example the distinction between audio and video call is not always possible if the call was initiated from the web interface. In this case the call always starts with audio only but video can be enabled at any time. On the left you see the outgoing call screen "ringing". On then right you see the incoming call screen, which indicated at the top that it is an audio call “Nextcloud Talk voice call”. | image

### Notes

* Distinction between audio and video call is not always possible. For example from the web interface a call is always started with audio only but video can be enabled at any time.

### Acceptance criteria

* The incoming call screen should indicate whether it is a audio or video call.
* The incoming call screen should indicate who is calling and whether it is an incoming or outdoing call.
* The call screen on both sides should indicate the current status, such as “calling”, “ringing”, “reconnection”, “connected”.

Ref https://github.com/nextcloud/talk-android/issues/956
Ref https://github.com/nextcloud/talk-ios/issues/465

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #4931

### CODE REVIEW

**Comment by nickvergessen** [comment]
Currently clients (web and ios tested) seem to always join the call without a parameter so all are added as `7` which is connected+audio+video

File: lib/Migration/Version11001Date20210211111527.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+namespace OCA\Talk\Migration;
+
+use Closure;
+use Doctrine\DBAL\Types\Types;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+/**
+ * Auto-generated migration step: Please modify to your needs!
```
Comment (PVince81) [inline lib/Migration/Version11001Date20210211111527.php:35]: stray comment

File: lib/Migration/Version11001Date20210211111527.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		$table = $schema->getTable('talk_rooms');
+		if (!$table->hasColumn('call_flag')) {
+
+			// Auto increment id
```
Comment (PVince81) [inline lib/Migration/Version11001Date20210211111527.php:51]: stray comment ?

**Review by PVince81 — APPROVED** [review]
Looks fine :+1: 

See minor comments

**Comment by nickvergessen** [comment]
Fixed the comments and rebased on the fixed integration tests

**Pull request merged**